ddl/ddl_api.go

```diff
@@ -775,10 +784,38 @@ func buildColumnsAndConstraints(
 	return cols, constraints, nil
 }
 
+func getDefaultCollationForUTF8MB4(sessVars *variable.SessionVars, cs string) (string, error) {
```

Comment:
IMHO it's weird to allow a `cs` parameter for `getDefaultCollationForUTF8MB4` and it returns nothing when `cs != 'utf8mb4'`.

For example, the [code above](https://github.com/zimulala/tidb/blob/2de36755d11812c0a99ce893871370895c92e5e8/ddl/ddl_api.go#L150-L151) confused me at the first glance:

```
	if !explicitCollation && explicitCharset {
		coll, err := getDefaultCollationForUTF8MB4(ctx.GetSessionVars(), charsetOpt.Chs)
```

Could you consider removing this new function and just merge the logic to `charset.GetDefaultCollation`?

Reply:
`charset. GetDefaultCollation` places many, this function(`getDefaultCollationForUTF8MB4`) is used and the function affects only DDL operations. So you can't put this function in